### PR TITLE
Avoid to emit duplicates for enum fileds

### PR DESCRIPTION
**Pull request by andreaTP**
**What this PR does / why we need it**:

The enum "ciphers" contains duplicated entries, since it's Json Schema it should have all unique elements:
https://json-schema.org/understanding-json-schema/reference/enum

**Special notes for your reviewer**:

The bug has been discovered by @iocanel in https://github.com/fabric8io/kubernetes-client/pull/5611

**Release note**:

```release-note
Fixed incorrect generation of ciphers to prevent duplicated enum values.
```

### CODE REVIEW

File: pkg/components/components_test.go
Comment (oshoval) [inline pkg/components/components_test.go:111]: We can optimize a lil, wdyt ?
```
for i, vi := range stringCiphers {
    for j := i + 1; j < len(stringCiphers); j++ {
```

**Comment by oshoval** [comment]
We have also this one in the CR
https://github.com/kubevirt/cluster-network-addons-operator/blob/3a2b93b4259772257ead6c28ba748936da28e0bf/pkg/network/kubemacpool.go#L126

maybe worth to also dedup it please ? (and then also to add a test for it `pkg/network/tlsSecurityProfile_test.go`)

**Comment by andreaTP** [comment]
Thanks for the review @oshoval , I'll go through it soon.

File: pkg/components/components_test.go
Comment (andreaTP) [reply pkg/components/components_test.go:111]: ok 👍

**Comment by andreaTP** [comment]
@oshoval should be ready for review again, thanks for the hints 👍

File: pkg/network/tlsSecurityProfile_test.go
Comment (oshoval) [inline pkg/network/tlsSecurityProfile_test.go:79]: can you please move this line right next to the for ?

File: pkg/network/tlsSecurityProfile.go
Comment (oshoval) [inline pkg/network/tlsSecurityProfile.go:23]: no need comments please
they tend to age, and code is self explanatory here

File: pkg/components/components_test.go
Comment (oshoval) [inline pkg/components/components_test.go:110]: `Expect(vi).ToNot(Equal(stringCiphers[j]))`
wdyt ?
also on https://github.com/kubevirt/cluster-network-addons-operator/pull/1748/files#diff-7aaa98f5da89ba48b6fd676ea6f71cc8791dfc21c34d61e7ca41631aa9fd3535R82 please if so

**Review by oshoval — COMMENTED** [review]
Thanks
just little nits

"I hope this is the only place where it should be fixed, but I'm having trouble compiling the entire project on my machine (Mac)." 
can be removed from PR desc

File: pkg/network/tlsSecurityProfile.go
Comment (andreaTP) [reply pkg/network/tlsSecurityProfile.go:23]: removed

**Comment by andreaTP** [comment]
I think I have addressed all the notes now, should I squash the commits? any more comments? 🙂

**Comment by oshoval** [comment]
Thanks a lot, sorry for the nit picking
ye squashing sounds great in this case
beside that lgtm :)

**Comment by andreaTP** [comment]
Squashed the commits 👍

**Comment by oshoval** [comment]
Thanks

/lgtm
/retest

@qinqon do you want to take a look ?

**Comment by oshoval** [comment]
those seems to be flakes due to quay or so, anyhow not relevant to this PR
once it is lgtm approve it will be auto restested

**Comment by andreaTP** [comment]
I think that the CI failures are unrelated to the changes in this PR.

**Comment by oshoval** [comment]
You totally right

@qinqon can you please review ?

**Comment by andreaTP** [comment]
Ping, anything should I do on my side to finalize this one?

**Comment by oshoval** [comment]
just need review of @qinqon please
/retest

File: pkg/components/components.go
Comment (qinqon) [inline pkg/components/components.go:1047]: Do the json "encoding" at the end we don't need to store double quotes at the set

```suggestion
				if m[c] {
					continue
				}
				suites = append(suites, extv1.JSON{Raw: []byte(fmt.Sprintf(`"%s"`, c))})
```

**Review by qinqon — CHANGES_REQUESTED** [review]
Small nit

File: pkg/network/tlsSecurityProfile_test.go
Comment (qinqon) [inline pkg/network/tlsSecurityProfile_test.go:83]: We should just compare with the exact slice using gomega matchers, dulication at test is worth it.
```suggestion
			ciphers, _ := SelectCipherSuitesAndMinTLSVersion(profile)
			Expect(ciphers).To(ConsistOf("foo", "bar"))
		
```

File: pkg/components/components_test.go
Comment (qinqon) [inline pkg/components/components_test.go:99]: Where are the duplicated ciphers ?

**Review by qinqon — CHANGES_REQUESTED** [review]

File: pkg/network/tlsSecurityProfile_test.go
Comment (andreaTP) [reply pkg/network/tlsSecurityProfile_test.go:83]: I differ here, and unrolling loops in tests is not worth it.
In this specific case, the reason is that any time you modify/change/add or remove a cipher, you add a completely redundant task to update the test.

I do believe that this is unnecessary and unneeded maintenance, would you mind expanding on why it would be any better?

File: pkg/components/components_test.go
Comment (andreaTP) [reply pkg/components/components_test.go:99]: Here for example:
https://github.com/kubevirt/cluster-network-addons-operator/blob/3a2b93b4259772257ead6c28ba748936da28e0bf/manifests/cluster-network-addons/0.90.1/network-addons-config.crd.yaml#L1523-L1537

Comment (qinqon) [reply pkg/components/components_test.go:99]: What I mean is that we should mock those, we are not really enforcing duplicated on the test.

File: pkg/network/tlsSecurityProfile_test.go
Comment (qinqon) [reply pkg/network/tlsSecurityProfile_test.go:83]: ok, fair enough.

File: pkg/components/components_test.go
Comment (andreaTP) [reply pkg/components/components_test.go:99]: The goal of this test(and this PR overall) is to ensure we don't emit duplicates in the "ciphers" enums, the fact that they do exist or not in the original slice is almost an implementation detail.

If you want to do mock testing on something this small, please go ahead, you can PR my branch, fix on top or do whatever you prefer.

Comment (qinqon) [reply pkg/components/components_test.go:99]: Ok

**Review by qinqon — APPROVED** [review]
/lgtm
/approve

**Comment by qinqon** [comment]
/hold

@andreaTP please squash the commits.

**Comment by andreaTP** [comment]
@qinqon squashed the commits, rebased on latest `main` ...

**Comment by oshoval** [comment]
Thanks

**Pull request merged**
